contracts/test/CreateOffchainAssetReceiptVaultFactory.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: CAL
+pragma solidity =0.8.17;
+
+import {OffchainAssetReceiptVault} from "../vault/offchainAsset/OffchainAssetReceiptVault.sol";
+import {ReceiptFactory} from "../vault/receipt/ReceiptFactory.sol";
+import {ReceiptVaultFactoryConfig} from "../vault/receipt/ReceiptVaultFactory.sol";
+import {OffchainAssetReceiptVaultFactory} from "../vault/offchainAsset/OffchainAssetReceiptVaultFactory.sol";
+
+abstract contract CreateOffchainAssetReceiptVaultFactory {
+    OffchainAssetReceiptVault public implementation;
```

Comment:
these should be immutable if they're built in the constructor

there's also no reason for any of them to be public, they should all be internal